tools/boilerplate.py

```diff
@@ -137,155 +144,102 @@ def boilerplate_gen():
         'semilogx',
         'semilogy',
         'specgram',
-        # 'spy',
+        'spy',
         'stackplot',
         'stem',
         'step',
         'streamplot',
+        'table',
+        'text',
+        'tick_params',
+        'ticklabel_format',
         'tricontour',
         'tricontourf',
         'tripcolor',
         'triplot',
         'violinplot',
         'vlines',
         'xcorr',
-        'barbs',
-    )
-
-    _misccommands = (
-        'cla',
-        'grid',
-        'legend',
-        'table',
-        'text',
-        'annotate',
-        'ticklabel_format',
-        'locator_params',
-        'tick_params',
-        'margins',
-        'autoscale',
     )
 
     cmappable = {
-        'contour': 'if %(ret)s._A is not None: sci(%(ret)s)',
-        'contourf': 'if %(ret)s._A is not None: sci(%(ret)s)',
-        'hexbin': 'sci(%(ret)s)',
-        'scatter': 'sci(%(ret)s)',
-        'pcolor': 'sci(%(ret)s)',
-        'pcolormesh': 'sci(%(ret)s)',
-        'hist2d': 'sci(%(ret)s[-1])',
-        'imshow': 'sci(%(ret)s)',
-        #'spy'    :  'sci(%(ret)s)',  ### may return image or Line2D
-        'quiver': 'sci(%(ret)s)',
-        'specgram': 'sci(%(ret)s[-1])',
-        'streamplot': 'sci(%(ret)s.lines)',
-        'tricontour': 'if %(ret)s._A is not None: sci(%(ret)s)',
-        'tricontourf': 'if %(ret)s._A is not None: sci(%(ret)s)',
-        'tripcolor': 'sci(%(ret)s)',
+        'contour': 'if __ret._A is not None: sci(__ret)',
+        'contourf': 'if __ret._A is not None: sci(__ret)',
+        'hexbin': 'sci(__ret)',
+        'scatter': 'sci(__ret)',
+        'pcolor': 'sci(__ret)',
+        'pcolormesh': 'sci(__ret)',
+        'hist2d': 'sci(__ret[-1])',
+        'imshow': 'sci(__ret)',
+        'spy': 'if isinstance(ret, cm.ScalarMappable): sci(__ret)',
+        'quiver': 'sci(__ret)',
+        'specgram': 'sci(__ret[-1])',
+        'streamplot': 'sci(__ret.lines)',
+        'tricontour': 'if __ret._A is not None: sci(__ret)',
+        'tricontourf': 'if __ret._A is not None: sci(__ret)',
+        'tripcolor': 'sci(__ret)',
     }
 
-    def format_value(value):
+    class value_formatter:
```

Comment:
Why make this a class?

Reply:
because str(parameter) calls repr(default), so that's how we override its output?

Reply:
ah, understand now.